samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -249,6 +251,44 @@ public void waitForFinish() {
     }
   }
 
+
+  private void createIntermediateStreams(List<StreamSpec> intStreams) throws Exception {
+    boolean streamsExist = getStreamManager().checkIfStreamsExist(intStreams);
```

Comment:
Why don't we get the subset of `intStreams` that are not created and only try creating with them? In case of stream already exists, we do the work twice; once for checking if it exists and other time during the actual creation which would again result in no op.